pydm/widgets/slider.py

```diff
@@ -26,41 +26,209 @@
+        """
+        slider_pos = self.getSliderPosition()
+
+        if self.orientation() == Qt.Horizontal:
+            handle_pos = slider_pos
+            click_pos = event.pos().x()
+        else:
+            handle_pos = slider_pos
+            click_pos = event.pos().y()
+
+        return handle_pos, click_pos
+
+    def shouldIncrement(self, handle_pos, click_pos):
+        """
+        Determine whether the slider value should be incremented based on positions.
+
+        Parameters
+        ----------
+        handle_pos : float
+            The position of the slider handle.
+        click_pos : float
+            The position where the user clicked.
+
+        Returns
+        -------
+        bool
+            True if the slider value should be incremented, False otherwise.
+        """
+        if self.orientation() == Qt.Horizontal:
+            return click_pos > handle_pos
+        else:
+            return click_pos < handle_pos
+
+    def getHandleSize(self):
+        """
+        Compute the size of the slider handle.
+
+        Returns
+        -------
+        QSize
+            The size of the slider handle.
+        """
+        handle_length = 20  # Fixed length for the handle
+        if self.orientation() == Qt.Horizontal:
+            return QSize(handle_length, self.height() // 2)
+        else:
+            return QSize(self.width() // 2, handle_length)
+
+    def getSliderLength(self):
+        """
+        Calculate the usable length of the slider track excluding the handle size.
+
+        Returns
+        -------
+        float
+            The length of the slider track.
+        """
+        handle_size = self.getHandleSize()
+        if self.orientation() == Qt.Horizontal:
+            return self.width() - handle_size.width()
+        else:
+            return self.height() - handle_size.height()
+
+    def getSliderPosition(self):
+        """
+        Compute the position of the slider handle along the track.
+
+        Returns
+        -------
+        float
+            The position of the slider handle along the slider track.
+        """
+        slider_min = self.minimum()
+        slider_max = self.maximum()
+        slider_range = slider_max - slider_min
+        slider_value = self.value() - slider_min
+
+        if slider_range == 0:
+            proportion = 0
+        else:
+            proportion = slider_value / slider_range
 
-    def handleWidth(self):
         if self.orientation() == Qt.Horizontal:
-            return self.style().pixelMetric(self.style().PM_SliderLength, None, self)
+            slider_length = self.getSliderLength()
+            return proportion * slider_length + self.getHandleSize().width() / 2
         else:
-            return self.style().pixelMetric(self.style().PM_SliderThickness, None, self)
+            slider_length = self.getSliderLength()
+            return (1 - proportion) * slider_length + self.getHandleSize().height() / 2
```

Comment:
add comment here about why '(1 - proportion)'?

its because proportion = 0 results in 1, which means largest value is at bottom of slider right?